state.py

```diff
@@ -1,5 +1,62 @@
 from type_wrapper import Integer, Value
 from rpython.rlib import jit
+from rpython.rtyper.lltypesystem import rffi
+
+import llvm_wrapper as llwrap
+
+class NoSuchVariableException(Exception):
+    pass
+
+class InterpreterState(object):
+
+    def __init__(self, module=None, frame=None, global_state=None, last_block=None):
+        #self.update(module, frame, global_state, last_block)
+        self.module = module
+        self.current_frame = frame
+        self.global_state = global_state
+        self.last_block = last_block
+
+    def update(self, module=None, frame=None, global_state=None, last_block=None):
+        if module:
+            self.module = module
+        if frame:
+            self.current_frame = frame
+        if global_state:
+            self.global_state = global_state
+        if last_block:
+            self.last_block = last_block
+
+    def lookup_var(self, var, ignore_err=False):
+        ''' Returns the value of a variable. First checks if the Variable
+            is a Constant; if it is not an LLVM constant, the
+            local variable dictionary is checked, followed by the global
+            variable dictionary. '''
+
+        from llvm_objects import Constant
+        if isinstance(var, Constant):
+            return var.content
+        elif self.current_frame.has_key(var.addr):
+            return self.current_frame.get_variable(var.addr)
+        elif self.global_state and self.global_state.has_key(var.addr):
+            return self.global_state.get_variable(var.addr)
+        elif not ignore_err:
+            print "[ERROR]: Unknown variable. Exiting."
+            error_str = rffi.charp2str(llwrap.LLVMPrintValueToString(var.l_value))
+            raise NoSuchVariableException(error_str)
```

Comment:
Should `lookup_var` define in its description that it returns `None` if `ignore_err=False`?